Спринт №12

Keithcutt opened the pull request:
Добавлена реализация переключения темной темы и истории поиска.

Keithcutt committed:
Домашнее задание №12

Попытка реализовать историю поиска

Keithcutt committed:
Домашнее задание №12

Вторая попытка реализовать историю поиска

Keithcutt committed:
Домашнее задание №12

Третья (как будто бы удачная) попытка реализовать историю поиска

Nick-review commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -0,0 +1,47 @@
+        private const val DARK_THEME = "key_for_theme_switcher"
+    }
+
+    var darkTheme = false
```
тут можно указать модификатор private set чтобы инкапсулировать изменения внутрь класса

Nick-review commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -50,54 +61,61 @@ class SearchActivity : AppCompatActivity() {
             inputMethodManager?.hideSoftInputFromWindow(it.windowToken, 0)
         }
 
+        val sharedPreferences = getSharedPreferences(SEARCH_HISTORY_PREFERENCES, MODE_PRIVATE)
```
Можно разбить метод onCreate на функции поменьше, а то он обрастает функционалом и его становится сложно читать

Nick-review commented on app/src/main/java/com/example/playlistmaker/SearchAdapter.kt:
```diff
@@ -1,21 +1,30 @@
 
     override fun getItemCount(): Int = tracks.size
+
+    fun interface TrackToSave {
```
Как-то сложновато для такого простого действия (обработка клика), можно на вход адаптеру передать просто лямбду обработчик кликов

Keithcutt committed:
Домашнее задание №12. Четвертая попытка.

1. Исправлены критические замечания по отображению истории поиска.

2. Переменная darkTheme теперь private. Для доступа этому параметру используется функиция-геттер.

3. Передаваемая в адаптер лямбда-функция была упрощена.

4. Часть функционала метода onCreate (в SearchActivity) была вынесена в отдельные функции. В ревью хотел бы получить комментарии по тому насколько хорошо у меня это получилось и что еще следовало бы вынести в отдельные функции. Я попробовал вынести создание и прикрепление TextWatchera, но это поламало функционал, поэтому вернул как было.

Keithcutt committed:
Домашнее задание №12. Пятая попытка.

1. Исправлены критические замечания по отображению истории поиска.

Keithcutt committed:
Домашнее задание №12. Пятая попытка.

Keithcutt committed:
Домашнее задание №12. Седьмая попытка.

Keithcutt committed:
Домашнее задание №12.

Не помню как следовало поступать в случае, когда запрос состоит из строки пробелов ("   "), сделал чтобы поиск не осуществлялся.

Все критические замечания вроде как исправил.

Keithcutt committed:
Домашнее задание №12.

Теперь ресайклер поиска всегда стартует с самого начала списка.

Nick-review commented on .idea/deploymentTargetDropDown.xml:
Всю папку .idea лучше занести в файл .gitignore, чтобы не заливать это в репозиторий, т.к. это индивидуальные настройки вашей среды разработки

Keithcutt committed:
Папка .idea была занесена в gitignore и удалена из git.
